pkg/controller/gather_commands.go

```diff
@@ -409,80 +421,91 @@ func wasDataProcessed(ctx context.Context,
-			requestRetryCounter++
-			return false, nil
+			return false, requestRetry(retryCounter, resp.StatusCode, delay)
 		}
 
-		// Process the successful response
-		if resp.Body == nil || resp.Body == http.NoBody {
-			return false, nil
-		}
+		return processSuccessfulResponse(resp.Body, retryCounter, delay)
+	})
+	if err != nil {
+		return false, err
+	}
 
-		data, err := io.ReadAll(resp.Body)
-		if err != nil {
-			return false, err
-		}
+	return true, nil
+}
 
-		var processingResp dataStatus
-		err = json.Unmarshal(data, &processingResp)
-		if err != nil {
-			return false, err
-		}
+// networkRetry is used to retry when there is a network issue that caused failure
+func networkRetry(retryCounter *retryCounter, err error, delay time.Duration) error {
+	if retryCounter.network >= retryCounter.max {
+		return fmt.Errorf("failed to check processing status after %d retries: %w", retryCounter.network, err)
+	}
+	klog.Infof("Network error when checking processing status: %v, retry %d/%d in %s",
+		err, retryCounter.network+1, retryCounter.max, delay)
+	retryCounter.network++
+	return nil
+}
 
-		dataProcessed = (processingResp.Status == "processed")
-
-		// If data is not processed yet, retry 3 times before failing
-		if !dataProcessed {
-			if statusRetryCounter >= numberOfStatusQueryRetries {
-				klog.Infof("Data status is %q after %d retries, stopping poll", processingResp.Status, statusRetryCounter)
-				return false, fmt.Errorf("data processing status is %q after %d retries, stopping poll", processingResp.Status, statusRetryCounter)
-			}
-			klog.Infof("Data status is %q, retry %d/%d in %s",
-				processingResp.Status, statusRetryCounter+1, numberOfStatusQueryRetries, delay)
-			statusRetryCounter++
-			return false, nil
-		}
+// requestRetry is used to retry an http request when the response != 200
+func requestRetry(retryCounter *retryCounter, respStatusCode int, delay time.Duration) error {
+	if retryCounter.request >= retryCounter.max {
+		return fmt.Errorf("HTTP status message: %s", http.StatusText(respStatusCode))
+	}
+	klog.Infof("Received HTTP status code %d, retry %d/%d in %s",
+		respStatusCode, retryCounter.request+1, retryCounter.max, delay)
+	retryCounter.request++
+	return nil
+}
 
-		return true, nil
-	})
+// processSuccessfulResponse is used to process response body and if data is not processed, it retries 3 times
+// Returns true if the data was successfully processed, false otherwise
+func processSuccessfulResponse(respBody io.ReadCloser, retryCounter *retryCounter, delay time.Duration) (bool, error) {
+	if respBody == nil || respBody == http.NoBody {
+		return false, nil
+	}
+
+	data, err := io.ReadAll(respBody)
+	if err != nil {
+		return false, err
+	}
+
+	var processingResp dataStatus
+	err = json.Unmarshal(data, &processingResp)
 	if err != nil {
 		return false, err
 	}
 
-	return dataProcessed, nil
+	// If data is not processed yet, retry 3 times before failing
```

Comment:
I saw this `3` value also in the documentation of `processSuccessfulResponse` but I don't see it anywhere. Could this be the `retryCounter.max` value?